receiver.go

```diff
@@ -154,11 +155,14 @@ func (r *receiver) receiveDatagram(l int) (int, *net.UDPAddr, error) {
 		if err != nil {
 			return 0, nil, err
 		}
-		// TODO: compare addr here?
+		if !addr.IP.Equal(r.addr.IP) || (r.tid != 0 && addr.Port != r.tid) {
```

Comment:
`r.tid != 0` is because of the fact we do not know the remote port (TID) before first packet comes so we should skip the check for the first packet?

That looks good.

Maybe the only minor change I would do is to make it a little bit more step by step:

```go
if !ip-matches {
  continue
}
if addr.Port != r.tid {
  if r.tid == 0 {
    // Skip the TID check for the first packet
    r.tid = addr.Port
  } else {
    continue
  }
}
```
It is more verbose, but a bit easier to understand the logic, on fast path (most of the iterations) `r.tid != 0` comparison is skipped and `r.tid` is assigned only once.

But. It might be just me, it looks good to me like it is now too.